plugin/evm/vm.go

```diff
@@ -674,7 +672,55 @@ func (vm *VM) initializeChain(lastAcceptedHash common.Hash) error {
 // initializeStateSyncClient initializes the client for performing state sync.
 // If state sync is disabled, this function will wipe any ongoing summary from
 // disk to ensure that we do not continue syncing from an invalid snapshot.
-func (vm *VM) initializeStateSyncClient(lastAcceptedHeight uint64) error {
+func (vm *VM) initializeStateSync(lastAcceptedHeight uint64) error {
+	// Create standalone EVM TrieDB (read only) for serving leafs requests.
+	// We create a standalone TrieDB here, so that it has a standalone cache from the one
+	// used by the node when processing blocks.
+	evmTrieDB := triedb.NewDatabase(
+		vm.chaindb,
+		&triedb.Config{
+			DBOverride: hashdb.Config{
+				CleanCacheSize: vm.config.StateSyncServerTrieCache * units.MiB,
+			}.BackendConstructor,
+		},
+	)
+	var leafHandlerConfigs []*extension.LeafRequestConfig
+	syncStats := handlerstats.GetOrRegisterHandlerStats(metrics.Enabled)
+	// register default leaf request handler for state trie
+	leafHandlerConfigs = append(leafHandlerConfigs, &extension.LeafRequestConfig{
+		LeafType:   message.StateTrieNode,
+		MetricName: "sync_state_trie_leaves",
+		Handler: handlers.NewLeafsRequestHandler(evmTrieDB,
+			message.StateTrieKeyLength,
+			vm.blockChain, vm.networkCodec,
+			syncStats,
+		),
+	})
+	leafHandler := atomicsync.NewAtomicLeafHandler(vm.atomicBackend.AtomicTrie().TrieDB(), atomicstate.TrieKeyLength, vm.networkCodec)
+	leafHandlerConfigs = append(leafHandlerConfigs, &extension.LeafRequestConfig{
+		LeafType:   atomicsync.AtomicTrieNode,
+		MetricName: "sync_atomic_trie_leaves",
+		Handler:    leafHandler,
+	})
+
+	leafHandlers := make(LeafHandlers, len(leafHandlerConfigs))
+	for _, leafConfig := range leafHandlerConfigs {
+		if _, exists := leafHandlers[leafConfig.LeafType]; exists {
+			return fmt.Errorf("duplicate leaf type %v", leafConfig.LeafType)
+		}
+		leafHandlers[leafConfig.LeafType] = leafConfig.Handler
+	}
```

Comment:
```suggestion
	leafHandlers := make(LeafHandlers, 2) // 2 default leaf handlers for state trie and atomic trie
	syncStats := handlerstats.GetOrRegisterHandlerStats(metrics.Enabled)

	// Register the leafs request handler for state trie
	leafHandlers[message.StateTrieNode] = handlers.NewLeafsRequestHandler(evmTrieDB,
		message.StateTrieKeyLength,
		vm.blockChain, vm.networkCodec,
		syncStats,
	)

	// Register the atomic trie leafs request handler
	leafHandlers[atomicsync.AtomicTrieNode] = atomicsync.NewAtomicLeafHandler(vm.atomicBackend.AtomicTrie().TrieDB(), atomicstate.TrieKeyLength, vm.networkCodec)
```

IIUC, this could simplify this initialization code.

Reply:
I kept this a bit "ugly" with the intention that we can register leaf configs externally (like in #885), these will be changed at the end.

Reply:
removed in https://github.com/ava-labs/coreth/pull/985

Reply:
Sorry I don't quite follow...I may be missing something here.

Why do we create handlers in `leafHandlerConfigs` only to then loop over it to assign into `leafHandlers`? It looks like that pattern still exists in #985 also